eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -114,6 +115,31 @@ object InteractiveTxBuilder {
     val minNextFeerate: FeeratePerKw = targetFeerate * 25 / 24
   }
 
+  // @formatter:off
+  sealed trait Purpose {
+    def previousLocalBalance: MilliSatoshi = 0.msat
+    def previousRemoteBalance: MilliSatoshi = 0.msat
+    def commitmentIndex: Long = 0
```

Comment:
I'm not a big fan of having default values in traits, it's easier to check for correctness when each purpose explicitly sets those values, otherwise a future purpose may incorrectly inherit invalid default values.